File: Association Slope/Hash/Q1_Two Sum/Q1_TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # Dictionary to store number -> index
        seen = {}
        
        # Loop through array
        for i, num in enumerate(nums):
            # Calculate the number we need to find
            complement = target - num
            
            # If complement already exists in dictionary, return indices
            if complement in seen:
                return [seen[complement], i]
            
            # Otherwise, store current number with its index
            seen[num] = i

File: Association Slope/Hash/Q1_Two Sum/Q1_TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>(); // Store value -> index
        
        for (int i = 0; i < nums.length; i++) 
        {
            int complement = target - nums[i]; // Find the required pair
            
            if (map.containsKey(complement)) {
                return new int[]{map.get(complement), i}; // Found the pair
            }
            
            map.put(nums[i], i); // Store number and its index
        }
        
        return new int[]{-1, -1}; // No solution found (edge case)
    }
}
